File: android/lca/src/main/java/com/dreampany/lca/data/source/room/RoomPriceDataSource.java
/*
package com.dreampany.lca.data.source.room;

import com.dreampany.lca.data.misc.PriceMapper;
import com.dreampany.lca.data.model.Price;
import com.dreampany.lca.data.source.api.PriceDataSource;

import java.util.List;

import javax.inject.Singleton;

import com.dreampany.lca.data.source.dao.PriceDao;
import io.reactivex.Maybe;

*/
/**
 * Created by Hawladar Roman on 7/23/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 *//*

@Singleton
public class RoomPriceDataSource implements PriceDataSource {

    private final PriceMapper mapper;
    private final PriceDao dao;

    public RoomPriceDataSource(PriceMapper mapper,
                               PriceDao dao) {
        this.mapper = mapper;
        this.dao = dao;
    }

    @Override
    public boolean isEmpty() {
        return getCount() == 0;
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return Maybe.fromCallable(this::isEmpty);
    }

    @Override
    public int getCount() {
        return dao.getCount();
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return null;
    }

    @Override
    public boolean isExists(Price price) {
        return dao.getCount(price.getId()) > 0;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Price price) {
        return Maybe.fromCallable(() -> isExists(price));
    }

    @Override
    public long putItem(Price price) {
        return dao.insertOrReplace(price);
    }

    @Override
    public Maybe<Long> putItemRx(Price price) {
        return Maybe.fromCallable(() -> dao.insertOrReplace(price));
    }

    @Override
    public List<Long> putItems(List<Price> prices) {
        return null;
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<Price> prices) {
        return null;
    }

    @Override
    public int delete(Price price) {
        return 0;
    }

    @Override
    public Maybe<Integer> deleteRx(Price price) {
        return null;
    }

    @Override
    public List<Long> delete(List<Price> prices) {
        return null;
    }

    @Override
    public Maybe<List<Long>> deleteRx(List<Price> prices) {
        return null;
    }

    @Override
    public List<Price> getItems() {
        return null;
    }

    @Override
    public Maybe<List<Price>> getItemsRx() {
        return null;
    }

    @Override
    public List<Price> getItems(int limit) {
        return null;
    }

    @Override
    public Maybe<List<Price>> getItemsRx(int limit) {
        return null;
    }

    @Override
    public Price getItem(String id) {
        return null;
    }

    @Override
    public Maybe<Price> getItemRx(String id) {
        return null;
    }
}
*/

File: android/lca/src/main/kotlin/com/dreampany/lca/data/source/room/RoomPriceDataSource.kt
package com.dreampany.lca.data.source.room

import com.dreampany.lca.data.misc.PriceMapper
import com.dreampany.lca.data.model.Price
import com.dreampany.lca.data.source.api.PriceDataSource
import com.dreampany.lca.data.source.dao.PriceDao
import io.reactivex.Maybe
import javax.inject.Singleton

/**
 * Created by roman on 2019-07-14
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Singleton
class RoomPriceDataSource constructor(
    val mapper: PriceMapper,
    val dao: PriceDao
) : PriceDataSource {

    override fun isEmpty(): Boolean {
        return getCount() == 0
    }

    override fun isEmptyRx(): Maybe<Boolean> {
        return Maybe.fromCallable({ this.isEmpty() })
    }

    override fun getCount(): Int {
        return dao.count
    }

    override fun getCountRx(): Maybe<Int> {
        return Maybe.empty()
    }

    override fun isExists(price: Price): Boolean {
        return dao.getCount(price.id) > 0
    }

    override fun isExistsRx(price: Price): Maybe<Boolean> {
        return Maybe.fromCallable { isExists(price) }
    }

    override fun putItem(price: Price): Long {
        return dao.insertOrReplace(price)
    }

    override fun putItemRx(price: Price): Maybe<Long> {
        return Maybe.fromCallable { dao.insertOrReplace(price) }
    }

    override fun putItems(prices: List<Price>): List<Long>? {
        return null
    }

   override fun putItemsRx(prices: List<Price>): Maybe<List<Long>> {
        return Maybe.empty()
    }

    override fun delete(price: Price): Int {
        return 0
    }

    override fun deleteRx(price: Price): Maybe<Int> {
        return Maybe.empty()
    }

   override fun delete(prices: List<Price>): List<Long>? {
        return null
    }

   override fun deleteRx(prices: List<Price>): Maybe<List<Long>> {
        return Maybe.empty()
    }

    override fun getItems(): List<Price>? {
        return null
    }

    override fun getItemsRx(): Maybe<List<Price>> {
        return Maybe.empty()
    }

    override fun getItems(limit: Int): List<Price>? {
        return null
    }

    override fun getItemsRx(limit: Int): Maybe<List<Price>> {
        return Maybe.empty()
    }

    override fun getItem(id: String): Price? {
        return null
    }

    override fun getItemRx(id: String): Maybe<Price> {
        return Maybe.empty()
    }
}